### ISSUE TITLE
unexpected indent behaviour in multiline boolean expression in 4.x

### ISSUE THREAD

**Issue by pavelkucera** [eslint/eslint#8914]
**Tell us about your environment**

* **ESLint Version:** 4.2.0
* **Node Version:** 8.1.3
* **npm Version:** 5.0.3

**What parser (default, Babel-ESLint, etc.) are you using?**
babel-eslint

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```json
{
    "parser": "babel-eslint",
    "env": {
        "es6": true,
        "node": true
    },
    "rules": {
        "indent": [
            "error",
            4,
            {
                "SwitchCase": 1,
                "MemberExpression": 1
            }
        ]
    }
}
```

</details>

**What did you do? Please include the actual source code causing the issue.**

```js
const skip = (request, response) => {
    return first
        || (
            second
            && third
        );
};
```

**What did you expect to happen?**
I expected the code to pass the indent rule with given configuration.

**What actually happened? Please include the actual, raw output from ESLint.**
```
/…/src/index.js
  5:1  error  Expected indentation of 8 spaces but found 12  indent
  7:1  error  Expected indentation of 4 spaces but found 8   indent
```

However, the proposed fix looks rather odd
```js
const skip = (request, response) => {
    return first
        || (
        second // no indentation here?
            && third // inconsistent indentation of the first and second part of the condition
    ); // inconsistent indentation of the opening and closing brace
};
```

Is this intended behaviour? Thank you.

**Comment by not-an-aardvark** [comment]
Thanks for reporting. This seems like a bug to me.

As a workaround, the rule won't complain if you do this:

```js
const skip = (request, response) => {
    return first
    || (
        second
        && third
    );
};
```

or this:

```js
const skip = (request, response) => {
    return first ||
        (
            second
            && third
        );
};
```

**Comment by pavelkucera** [comment]
Thank you a lot for a quick fix!
